File: leetcode-04/solutions/python3/0621.py
class Solution:
    def leastInterval(self, tasks: List[str], n: int) -> int:
        dict = [0] * 26
        for task in tasks:
            dict[ord(task) - ord('A')] += 1
        dict.sort()

        i = 25

        while i >= 0 and dict[i] == dict[25]:
            i -= 1

        return max(len(tasks), (dict[25] - 1) * (n + 1) + 25 - i)

File: leetcode-04/solutions/cpp/0621.cpp
class Solution {
 public:
  int leastInterval(vector<char>& tasks, int n) {
    vector<int> map(26);
    for (char task : tasks) map[task - 'A']++;
    sort(map.begin(), map.end());

    int i = 25;

    while (i >= 0 && map[i] == map[25]) i--;

    return max((int)tasks.size(), (map[25] - 1) * (n + 1) + 25 - i);
  }
};